.github/workflows/build-and-deploy.yml

```
@@ -121,6 +121,53 @@ jobs:
           maven-central-password: ${{ secrets.MAVEN_CENTRAL_PASSWORD }}
           signing-key: ${{ secrets.SIGNING_KEY }}
           signing-password: ${{ secrets.SIGNING_PASSWORD }}
+  test-multi-stage-deployment:
+    runs-on: ubuntu-latest
+    if: >-
+      !github.event.repository.fork
+      && (github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository)
+    steps:
+      - name: Compute the version of the target test project
+        id: versiontrick
+        shell: bash
+        run: |
+          # Idea: the regex matcher of Renovate keeps this string up to date automatically
+          # The version is extracted and used to access the correct version of the scripts
+          USES=$(cat <<TRICK_RENOVATE
+          - uses: DanySK/template-for-kotlin-multiplatform-projects@0.1.7
+          TRICK_RENOVATE
+          )
+          echo "Scripts update line: \"$USES\""
+          echo "Computed version: \"${USES#*@}\""
+          echo "version=${USES#*@}" >> $GITHUB_OUTPUT
+      - name: Checkout Template-for-Kotlin-Multiplatform-Projects ${{ steps.versiontrick.outputs.version }}
+        uses: actions/checkout@v3.3.0
+        with:
+          fetch-depth: '0'
+          path: 'kt-mp-multi-stage'
+          ref: "${{ steps.versiontrick.outputs.version }}"
+          repository: 'DanySK/Template-for-Kotlin-Multiplatform-Projects'
+          submodules: 'recursive'
+      - name: Checkout publish-on-central
+        uses: actions/checkout@v3.3.0
+        with:
+          path: 'publish-on-central'
+      - name: Dry-deploy
+        uses: DanySK/build-check-deploy-gradle-action@2.1.21
+        with:
+          build-command: true
+          check-command: true
+          deploy-command: |
+            COMMAND='./gradlew --include-build ../publish-on-central createStagingRepositoryOnMavenCentral --parallel'
+            $(echo "$COMMAND") || $(echo "$COMMAND") || $(echo "$COMMAND")
+            [[ -e build/staging-repo-ids.properties ]]
+            [[ $(cat build/staging-repo-ids.properties) =~ "^MavenCentral=(.*)$" ]]
```

```suggestion
            [[  "$(wc -l build/staging-repo-ids.properties)" ~= "^1 .*" ]]
            [[ "$(cat build/staging-repo-ids.properties)" =~ '^MavenCentral=\w*$' ]]
```

1. single quotes to avoid escaping issues
2. repo ids should be word characters (intercepts spurious endings)
3. always bash-expand into quoted environments
4. check that it is a one-liner in this test to avoid a multiline match

---

@nicolasfara that's a minor improvement. If it works, I'll enable autosquash.

---

I've noticed that the repo id should be like `orgdanilopianini-8124` so the regex `'^MavenCentral=\w*$'` do not match, could you confirm that?
I propose the following regex: `'^MavenCentral=.+-\d+$'`.

How does it sound?

---

sounds like `'^MavenCentral=\w+-\d+$'` would be a better pick, then (it would match `_-1`, but it does not seem too much of a concern)

---

Ok, pushed all the changes